File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int r_index = 0;
        int l_index = 0;

        for (l_index = 0; l_index< nums.length; l_index++) {

            if (nums[l_index] == 0) {
                for(r_index = l_index+1; r_index < nums.length; r_index++) {

                    if(nums[r_index] != 0) {
                        nums[l_index] = nums[r_index];
                        nums[r_index] = 0;
                        break;
                    }

                }
            }

        }

    }
}

File: 283-move-zeroes/move-zeroes.go
func moveZeroes(nums []int) {
	length := len(nums)

	for k := 0; k < length-1/2; k++ {
		for i := 0; i < length-1; i++ {
			if nums[i] == 0 && nums[i+1] != 0 {
				nums[i] = nums[i+1]
				nums[i+1] = 0
			}
		}
	}
}